### PR TITLE
Added index statistics to dataset.stats

### PR DESCRIPTION
**Pull request by minhtuev**
## What changes are proposed in this pull request?

Added an argument `include_indexes` so that we can return index statistics 

## How is this patch tested? If it is not, please explain why.

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart-video")

dataset.create_index("frames.detections.detections.label")

fo.pprint(dataset.get_index_information(include_size=True))

"""
{
    'id': {'v': 2, 'key': [('_id', 1)], 'size': 4096},
    'filepath': {'v': 2, 'key': [('filepath', 1)], 'size': 4096},
    'frames.id': {'v': 2, 'key': [('_id', 1)], 'size': 4096},
    'frames._sample_id_1_frame_number_1': {
        'v': 2,
        'key': [('_sample_id', 1), ('frame_number', 1)],
        'unique': True,
        'size': 4096,
    },
    'frames.detections.detections.label': {
        'v': 2,
        'key': [('detections.detections.label', 1)],
        'size': 28672,
    },
}
"""

fo.pprint(dataset.stats(include_indexes=True))

"""
{
    'samples_count': 10,
    'samples_bytes': 1820,
    'samples_size': '1.8KB',
    'frames_count': 1279,
    'frames_bytes': 2249818,
    'frames_size': '2.1MB',
    'indexes_count': 5,
    'indexes_bytes': 45056,
    'indexes_size': '44.0KB',
    'index_bytes': {
        'id': 4096,
        'filepath': 4096,
        'frames.id': 4096,
        'frames._sample_id_1_frame_number_1': 4096,
        'frames.detections.detections.label': 28672,
    },
    'index_sizes': {
        'id': '4.0KB',
        'filepath': '4.0KB',
        'frames.id': '4.0KB',
        'frames._sample_id_1_frame_number_1': '4.0KB',
        'frames.detections.detections.label': '28.0KB',
    },
    'total_bytes': 2296694,
    'total_size': '2.2MB',
}
"""
```

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

Users can now view index statistics using `dataset.stats(include_indexes=True)`

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- The `stats` method now includes an optional parameter to include index statistics, enhancing insights into the dataset structure.
	- The `get_index_information` method has been updated with a new parameter to optionally include index sizes, providing more comprehensive index information.
- **Tests**
	- Added a new test to validate the functionality of indexing within datasets, ensuring accurate creation and size retrieval for indices.

### CODE REVIEW

**Comment by minhtuev** [comment]
Examples:

@benjaminpkane : probably I should do some conversions for the index sizes to MB

```
In [1]: import fiftyone as fo
   ...: import fiftyone.zoo as foz

In [2]: dataset = fo.load_dataset("oi-v7full-10mm")

In [3]: dataset.stats(include_indexes=True)

Out[3]:
{'samples_count': 10000000,
 'samples_bytes': 156539929160,
 'samples_size': '145.8GB',
 'nindexes': 3,
 'totalIndexSize': 474648576,
 'indexSizes': {'_id_': 130613248,
  'filepath_1': 199389184,
  'detections.detections.str1_1': 144646144},
 'total_bytes': 156539929160,
 'total_size': '145.8GB'}
```

```
In [1]: import fiftyone as fo
   ...: import fiftyone.zoo as foz

In [2]: dataset = fo.load_dataset("oi-v7full-10mm")

In [3]: dataset.get_index_information(include_size=True)

Out[3]:
{'id': {'v': 2, 'key': [('_id', 1)], 'size': 130613248},
 'filepath': {'v': 2, 'key': [('filepath', 1)], 'size': 199389184},
 'detections.detections.str1': {'v': 2,
  'key': [('detections.detections.str1', 1)],
  'size': 144646144}}
  ```

File: fiftyone/core/collections.py [hunk 1]
```diff
@@ -9043,6 +9046,14 @@ def get_index_information(self):
         # Sample-level indexes
         fields_map = self._get_db_fields_map(reverse=True)
         sample_info = self._dataset._sample_collection.index_information()
+
+        if include_size:
+            dataset_stats = self._dataset.stats(include_indexes=True)
```
Comment (minhtuev) [inline fiftyone/core/collections.py:9051]: It is not ideal to do two calls to MongoDB to get the stats ~ still trying to figure out a better way :)

**Comment by minhtuev** [comment]
Converted from raw bytes to human-readable byte strings

```
{'samples_count': 10000000,
 'samples_bytes': 156539929160,
 'samples_size': '145.8GB',
 'nindexes': 3,
 'totalIndexSize': 474648576,
 'indexSizes': {'_id_': '124.6MB',
  'filepath_1': '190.2MB',
  'detections.detections.str1_1': '137.9MB'},
 'total_bytes': 156539929160,
 'total_size': '145.8GB'}

In [4]: dataset.get_index_information(include_size=True)
Out[4]:
{'id': {'v': 2, 'key': [('_id', 1)], 'size': '124.6MB'},
 'filepath': {'v': 2, 'key': [('filepath', 1)], 'size': '190.2MB'},
 'detections.detections.str1': {'v': 2,
  'key': [('detections.detections.str1', 1)],
  'size': '137.9MB'}}
```

**Review by brimoor — CHANGES_REQUESTED** [review]
@minhtuev note that the existing fields have this convention:

```
 'XXX_bytes': 156539929160,
 'XXX_size': '145.8GB',
```

The inclusion of `_size` was pretty cutesy on my part (helpful when working in repl), but `_bytes` is more suitable when actually trying to do something programmatic with the information.

So, for consistency with the existing format, how about this structure when `include_indexes=True`?

```
{
    'samples_count': 10000000,
    'samples_bytes': 156539929160,
    'samples_size': '145.8GB',
    'indexes_bytes': 474648576,
    'indexes_size': '452.7MB',
    'index_bytes': {
        'id': 130613248,
        'filepath': 199389184,
        'detections.detections.str1': 144646144,
    },
    'index_sizes': {
        'id': '124.6MB',
        'filepath': '190.2MB',
        'detections.detections.str1': '137.9MB',
    },
    'total_bytes': 156539929160,
    'total_size': '145.8GB',
}
```

Note that, for consistency with how `include_media=True` works, when `include_indexes=True` you should include the index sizes in `total_bytes`/`total_size`.

**Comment by minhtuev** [comment]
@brimoor : makes sense, done ✅ 

```
In [1]: import fiftyone as fo
   ...: import fiftyone.zoo as foz

In [2]: dataset = fo.load_dataset("oi-v7full-10mm")

In [3]: dataset.stats(include_indexes=True)
Out[3]:
{'samples_count': 10000000,
 'samples_bytes': 156539929160,
 'samples_size': '145.8GB',
 'num_indexes': 3,
 'indexes_bytes': 474648576,
 'indexes_sizes': '452.7MB',
 'index_bytes': {'_id_': 130613248,
  'filepath_1': 199389184,
  'detections.detections.str1_1': 144646144},
 'index_sizes': {'_id_': '124.6MB',
  'filepath_1': '190.2MB',
  'detections.detections.str1_1': '137.9MB'},
 'total_bytes': 156539929160,
 'total_size': '145.8GB'}

In [4]: dataset.get_index_information(include_size=True)
Out[4]:
{'id': {'v': 2, 'key': [('_id', 1)], 'size': '124.6MB', 'bytes': 130613248},
 'filepath': {'v': 2,
  'key': [('filepath', 1)],
  'size': '190.2MB',
  'bytes': 199389184},
 'detections.detections.str1': {'v': 2,
  'key': [('detections.detections.str1', 1)],
  'size': '137.9MB',
  'bytes': 144646144}}
```

File: fiftyone/core/collections.py [hunk 2]
```diff
@@ -9054,6 +9089,15 @@ def get_index_information(self):
             # Frame-level indexes
             fields_map = self._get_db_fields_map(frames=True, reverse=True)
             frame_info = self._dataset._frame_collection.index_information()
+
+            if include_size:
```
Comment (brimoor) [inline fiftyone/core/collections.py:9093]: FYI- added support for frame indexes on video datasets

**Review by brimoor — DISMISSED** [review]
LGTM

**Review by brimoor — APPROVED** [review]
LGTM

File: fiftyone/core/collections.py (hunk 2 above)
Comment (minhtuev) [reply fiftyone/core/collections.py:9093]: Aha!

**Pull request merged**